eth_account/_utils/structured_data/hashing.py

```diff
@@ -156,7 +156,7 @@ def get_array_dimensions(data):
 @to_tuple
 def flatten_multidimensional_array(array):
     for item in array:
-        if not isinstance(item, (list, tuple)):
+        if isinstance(item, (list, tuple)):
```

Comment:
@Bhargavasomu I *think* this was just a flipped logic bug. Does anything here look wrong to you?

Reply:
Hi @carver, it's been a long time since I wrote that code, and I do believe it is a flipped logic bug. Thank you for catching it. As far as the function definition and the usage goes, I don't see anything wrong with this change.